httpserver/healthcheck/healthcheck.go

```diff
@@ -56,8 +58,23 @@ func handlePprof(c *gin.Context) {
 	}
 }
 
-func (a *API) Handler() http.Handler {
-	return a.router
+func handleHealth(h *health.Health) func(*gin.Context) {
+	return func(c *gin.Context) {
+		ctx := c.Request.Context()
+		check := h.Measure(ctx)
+
+		data, err := json.Marshal(check)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+			return
+		}
+		code := http.StatusOK
+		if check.Status == health.StatusUnavailable {
+			o11y.AddField(ctx, "failures", check.Failures)
+			code = http.StatusServiceUnavailable
+		}
+		c.Data(code, "application/json", data)
```

Comment:
Can this not be a c.JSON, and ditch the explicit serialising above?

Reply:
Ah yeah it's a copy pasta